deployment/cdk/opensearch-service-migration/lib/network-stack.ts

```diff
@@ -61,35 +56,27 @@ export class NetworkStack extends Stack {
             });
         }
 
-        // If specified, these subnets will be selected to place the Domain nodes in. Otherwise, this is not provided
-        // to the Domain as it has existing behavior to select private subnets from a given VPC
-        if (props.vpcSubnetIds) {
-            const selectSubnets = this.vpc.selectSubnets({
-                subnetFilters: [SubnetFilter.byIds(props.vpcSubnetIds)]
-            })
-            this.domainSubnets = [selectSubnets]
-        }
+        if (!props.addOnMigrationDeployId) {
 
-        // Retrieve existing SGs to apply to VPC Domain endpoints
-        const securityGroups: ISecurityGroup[] = []
-        if (props.vpcSecurityGroupIds) {
-            for (let i = 0; i < props.vpcSecurityGroupIds.length; i++) {
-                securityGroups.push(SecurityGroup.fromLookupById(this, "domainSecurityGroup-" + i, props.vpcSecurityGroupIds[i]))
-            }
-        }
-        // Create a default SG which only allows members of this SG to access the Domain endpoints
-        const defaultSecurityGroup = new SecurityGroup(this, 'domainMigrationAccessSG', {
-            vpc: this.vpc,
-            allowAllOutbound: false,
-        });
-        defaultSecurityGroup.addIngressRule(defaultSecurityGroup, Port.allTraffic());
-        securityGroups.push(defaultSecurityGroup)
-        this.domainSecurityGroups = securityGroups
+            new StringParameter(this, 'SSMParameterVpcId', {
+                description: 'OpenSearch migration parameter for VPC id',
+                parameterName: `/migration/${props.stage}/${props.defaultDeployId}/vpcId`,
+                stringValue: this.vpc.vpcId
+            });
+
+            // Create a default SG which only allows members of this SG to access the Domain endpoints
+            const defaultSecurityGroup = new SecurityGroup(this, 'domainMigrationAccessSG', {
+                vpc: this.vpc,
+                allowAllOutbound: false,
+            });
+            defaultSecurityGroup.addIngressRule(defaultSecurityGroup, Port.allTraffic());
 
-        new CfnOutput(this, 'CopilotDomainSGExports', {
-            value: `export MIGRATION_DOMAIN_SG_ID=${defaultSecurityGroup.securityGroupId}`,
-            description: 'Domain Security Group created by CDK that is needed for Copilot container deployments',
-        });
+            new StringParameter(this, 'SSMParameterOpenSearchAccessGroupId', {
+                description: 'OpenSearch migration parameter for target OpenSearch access security group id',
+                parameterName: `/migration/${props.stage}/${props.defaultDeployId}/osAccessSecurityGroupId`,
+                stringValue: defaultSecurityGroup.securityGroupId
+            });
+        }
```

Comment:
1. A lot of the reason we are using SSM is to remove hard dependencies between stacks. While not really a concern for the network stack, we have taken the approach to limit stack exports as much as possible to allow doing things like replacing a target domain without having to take down all our other infrastructure as would be the case if the domain stack created any exports that were being used.
2. The concept of `defaultDeployId` and `deployId` roots from the multiple replay scenario where we want to allow additional replay infrastructure on an original(or default) deployment. The reason in some places you will see defaultDeployId used instead of deployId is to differentiate whether an additional replayer needs to create a unique resource or reuse an existing one. In the case of the security group here, we have opted to reuse the original security group for the domain that we created on the original deployment instead of creating a new unique security group for each additional deployment.